### PR TITLE
feat: support streaming method

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**
Unit test pass against local showcase server.

### CODE REVIEW

File: templates/typescript_gapic/package.json.njk [hunk 1]
```diff
@@ -11,15 +11,15 @@
...
+    "google-gax": "^1.5.0"
   },
   "devDependencies": {
     "@types/mocha": "^5.2.5",
     "@types/through2": "^2.0.34",
     "gts": "^0.9.0",
     "mocha": "^6.0.0",
     "typescript": "~3.5.3",
-    "google-gax": "^1.3.0"
+    "google-gax": "^1.5.0"
```
Comment (alexander-fenster) [inline templates/typescript_gapic/package.json.njk]: `dependencies` is something that the tool needs to work, `devDependencies` is something that the tool developers need (e.g. test frameworks). Gax (its protos) are needed to work, so it should be removed from `devDependencies`: just `dependencies` is enough.

File: templates/typescript_gapic/src/$version/$service_client.ts.njk [hunk 2]
```diff
@@ -174,23 +174,24 @@ export class {{ service.name }}Client {
...
+        stub => (...args: Array<{}>) => {
+          return stub[methodName].apply(stub, args);
+        },
+        (err: Error|null|undefined) => () => {
+          throw err;
+        });
+
       this._innerApiCalls[methodName] = gax.createApiCall(
-          {{ service.name.toCamelCase() }}Stub.then(
-              // @ts-ignore
```
Comment (alexander-fenster) [inline templates/typescript_gapic/src/$version/$service_client.ts.njk]: 🎉

File: typescript/test/testdata/echo_client_baseline.ts.txt [hunk 3]
```diff
@@ -227,4 +229,33 @@ export class EchoClient {
     options = options || {};
     return this._innerApiCalls.echo(request, options, callback);
   }
```
Comment (alexander-fenster) [inline typescript/test/testdata/echo_client_baseline.ts.txt]: Let's figure out how to add new lines between methods.

File: typescript/test/testdata/echo_client_baseline.ts.txt [hunk 4]
```diff
@@ -227,4 +229,33 @@ export class EchoClient {
...
+  expand(
+      request?: protos.google.showcase.v1beta1.IExpandRequest,
+      options?: gax.CallOptions):
+    gax.ServerStreamingCall{
```
Comment (alexander-fenster) [inline typescript/test/testdata/echo_client_baseline.ts.txt]: Missing space before `{`

File: typescript/test/testdata/echo_client_baseline.ts.txt [hunk 5]
```diff
@@ -227,4 +229,33 @@ export class EchoClient {
...
+    request = request || {}; 
+    options = options || {};
+    return this._innerApiCalls.expand(request, options);
+  }
+  collect(
+      options: gax.CallOptions|Callback<
```
Comment (alexander-fenster) [inline typescript/test/testdata/echo_client_baseline.ts.txt]: It seems there should be two overloads here:
```
collect(options: gax.CallOptions, callback: Callback<......>);
collect(callback: Callback<.......>);
```
and the function definition would then be
```
collect(optionsOrCallback: gax.CallOptions|Callback<......>, callback?: Callback<......>) {
   ...
}
```

Comment (alexander-fenster) [reply typescript/test/testdata/echo_client_baseline.ts.txt]: The difference is that the current overloads allow calling `collect(callback, options)` (reversed order) and it should be forbidden.

File: typescript/test/testdata/echo_client_baseline.ts.txt [hunk 6]
```diff
@@ -227,4 +229,33 @@ export class EchoClient {
...
+        protos.google.showcase.v1beta1.IEchoResponse,
+        protos.google.showcase.v1beta1.IEchoRequest|undefined, {}|undefined>,
+      callback: gax.CallOptions|Callback<
+        protos.google.showcase.v1beta1.IEchoResponse,
+        protos.google.showcase.v1beta1.IEchoRequest|undefined, {}|undefined>):
+    gax.ClientStreamingCall|void{
```
Comment (alexander-fenster) [inline typescript/test/testdata/echo_client_baseline.ts.txt]: why `void`? It must return a stream anyway.

**Comment by alexander-fenster** [comment]
Thank you, it's great! Left some minor comments.

**Comment by alexander-fenster** [comment]
It seems CircleCI did not run the tests on this PR, even though it should've done it. Would it be possible for you to trigger the CI build for this PR from your side? (I can't do it with your forked repo)

**Comment by xiaozhenliu-gg5** [comment]
Thanks Alex! will fork mainline/master as work base for next PR.

**Pull request merged**